### PR TITLE
chore: change consumer name max length match with route name

### PR DESCRIPTION
**Pull request by tao12345666333**
Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

### What this PR does / why we need it:

change consumer name's max length to match with route name.

ref:  https://github.com/apache/apisix-ingress-controller/pull/529

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by gxthrj — APPROVED** [review]
LGTM， But please see if the length adjustment will affect other modules @spacewander

**Review by spacewander — COMMENTED** [review]
If consumer name's max length needs to match with route name, we need to refer to the max length of route name's schema instead of using a hardcoded number.

**Comment by spacewander** [comment]
Will this change affect the dashboard? @nic-chen @juzhiyuan

**Comment by nic-chen** [comment]
> Will this change affect the dashboard? @nic-chen @juzhiyuan

Need to modify the length limit accordingly. We could update the dashboard after this PR merged.

**Comment by juzhiyuan** [comment]
Ok

File: apisix/schema_def.lua
Comment (tao12345666333) [inline apisix/schema_def.lua:623]:
```suggestion
            type = "string", minLength = 1, maxLength = rule_name_def.maxLength,
```

@spacewander Do you mean referencing this field separately?

**Review by spacewander — APPROVED** [review]

**Pull request merged**
